packages/core/src/render3/ng_module_ref.ts

```diff
@@ -23,6 +23,26 @@ import {getNgLocaleIdDef, getNgModuleDef} from './definition';
 import {setLocaleId} from './i18n/i18n_locale_id';
 import {maybeUnwrapFn} from './util/misc_utils';
 
+
+export function createNgModuleRef__PRE_R3__<T>(
+    ngModule: Type<T>, parentInjector?: Injector): NgModuleRef<T> {
+  throw new Error(`This API is Ivy-only and is not supported in ViewEngine`);
+}
+
+export function createNgModuleRef__POST_R3__<T>(
+    ngModule: Type<T>, parentInjector?: Injector): NgModuleRef<T> {
+  return new NgModuleRef<T>(ngModule, parentInjector ?? null);
+}
+
+/**
+ * Returns the NgModuleRef instance based on the NgModule class and parent injector provided.
```

Comment:
```suggestion
* Returns a new NgModuleRef instance based on the NgModule class and parent injector provided.
```
The current wording sounds like it might return a singleton. But it actually creates a new instance each time it is called.

Reply:
I think it might be worth, given that this is a public API, explaining what it means to create an NgModule without providing a parent injector. AFAICT it uses `getNullInjector()` instead.

Reply:
@petebacondarwin Wouldn't it also be valuable to have a method like `getNgModuleRefsById` to get a list of references (in most cases still just a single instance) of already instantiated modules and have direct access to their methods like `.destroy(): void`?

Reply:
Hmm, interestingly, I note that when creating multiple instances of an NgModule with the same `id`, only the last one to be created is stored in the registry. So `getNgModuleById()` would only ever return a single instance...

Reply:
@petebacondarwin Hmm, even when doing it fully programmatically? What happens with an already existed module instance? Is it simply overwritten in the registry even already used in an application?

Reply:
Oh hold on... `getNgModuleById()` returns the "type" of the NgModule, not any instance.

Reply:
So `getNgModuleRefsById()` would be something completely different...

Reply:
If the overwriting is the reality, then it should be the `getNgModuleRefById` method. I am targeting the easy availability to destroy a module instance to release the used memory even though a router creates this module instance.